evm/gas-service.js

```diff
@@ -0,0 +1,252 @@
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet } = require('./sign-utils');
+
+let failedChainUpdates = [];
+
+async function getGasUpdates(config, env, chain, destinationChains) {
+    const api = config.axelar.axelarscanApi;
+
+    validateParameters({
+        isNonEmptyStringArray: { destinationChains },
+    });
+
+    return Promise.all(
+        destinationChains.map(async (destinationChain) => {
+            const destinationConfig = config.chains[destinationChain];
+
+            if (!destinationConfig) {
+                printError(`Error: chain ${destinationChain} not found in config.`);
+                printError(`Skipping ${destinationChain}.`);
+                failedChainUpdates.push({ chain: chain.axelarId, destinationChain });
+                return null;
+            }
+
+            const { axelarId, onchainGasEstimate: { chainName = axelarId, gasEstimationType = 0, blobBaseFee = 0 } = {} } =
+                destinationConfig;
+
+            let data;
+
+            try {
+                data = await httpPost(`${api}/gmp/getFees`, {
+                    sourceChain: chain.axelarId,
+                    destinationChain: axelarId,
+                    sourceTokenAddress: AddressZero,
+                });
+            } catch (e) {
+                printError(`Error getting gas info for ${chain.axelarId} -> ${axelarId}`);
+                printError(e);
+                failedChainUpdates.push({ chain: chain.axelarId, destinationChain: axelarId });
+                return null;
+            }
+
+            const {
+                source_base_fee: sourceBaseFee,
+                source_token: {
+                    gas_price_in_units: { value: gasPrice },
+                    token_price: { usd: srcTokenPrice },
+                    decimals,
+                },
+                destination_native_token: {
+                    token_price: { usd: destinationTokenPrice },
+                },
+                execute_gas_multiplier: multiplier = 1.1,
+            } = data.result;
+
+            // sourceBaseFee * 10 ^ decimals
+            const axelarBaseFee = FixedNumber.from(parseFloat(sourceBaseFee).toFixed(10))
+                .mulUnsafe(FixedNumber.from(Math.pow(10, decimals).toFixed(10)))
+                .round();
+            // gasPrice * multiplier
+            const relativeGasPrice = FixedNumber.from(parseFloat(gasPrice))
+                .mulUnsafe(FixedNumber.from(parseFloat(multiplier).toFixed(10)))
+                .round();
+            // destinationTokenPrice / srcTokenPrice
+            const gasPriceRatio = FixedNumber.from(parseFloat(destinationTokenPrice).toFixed(10)).divUnsafe(
+                FixedNumber.from(parseFloat(srcTokenPrice).toFixed(10)),
+            );
+            // blobBaseFee * gasPriceRatio
+            const relativeBlobBaseFee = FixedNumber.from(parseFloat(blobBaseFee)).mulUnsafe(gasPriceRatio);
+
+            return {
+                chainName,
```

Comment:
it needs to be stored under both the `axelarId` and `ethereum`, since if the destination is ethereum, then `axelarId` version should use. But for L2s `ethereum` key is used